src/main/java/com/dataguise/saas/service/SchedulerService.java

```diff
@@ -0,0 +1,434 @@
+                    taskIdList.add(detectionTaskInstanceId);
+                }
+
+            } else if (schedulerDto.getScheduleType().equalsIgnoreCase(taskBuilderService.PROTECTION_TYPE)) {
+            /* One taskParamsDTO consists of one complete connection details.
+            In Masking create one protection task for each TaskParamsDTO. */
+                taskType = taskBuilderService.PROTECTION_TYPE;
+                for (TaskParamsDTO taskParamsDTO : taskParams) {
+                    DgMaskerTaskDTO protectionTask = protectionTaskService.constructProtectionTask(taskParamsDTO, schedulerDto.getSourceModule());
+                    System.out.println(protectionTask);
+                    Integer protectionTaskInstanceId = protectionTaskService.createTask(protectionTask, schedulerDto.getSourceModule());
+                    if (protectionTaskInstanceId != null) {
+                        taskIdList.add(protectionTaskInstanceId);
+                    }
+                }
+            }
+        } else {
+            if (schedulerDto.getScheduleType().equalsIgnoreCase(taskBuilderService.DETECTION_TYPE)) {
+                taskType = taskBuilderService.DETECTION_TYPE;
+                List<Integer> connectionIdList = new ArrayList<>();
+                for (TaskParamsDTO taskParamsDTO : taskParams) {
+                    connectionIdList.add(taskParamsDTO.getSourceId());
+                }
+                CloudTaskBean cloudDetectionTask = detectionTaskService.constructLFADetectionTask(
+                        schedulerDto.getScanLocations());
+                cloudDetectionTask.setClusterName(schedulerDto.getTaskParams().get(0).getSourceName());
+                Integer lfaDetectionTaskInstanceId = detectionTaskService.createLFATask(cloudDetectionTask);
+                if (lfaDetectionTaskInstanceId != null) {
+                    taskIdList.add(lfaDetectionTaskInstanceId);
+                }
+            }
+        }
+
+        DgTaskScheduleInfoBean taskScheduleBeanInfoBean = new DgTaskScheduleInfoBean();
+        taskScheduleBeanInfoBean.setId(0);
+        taskScheduleBeanInfoBean.setScheduleName(schedulerDto.getScheduleName());
+        taskScheduleBeanInfoBean.setScheduleInterval(schedulerDto.getScheduleInterval());
+        taskScheduleBeanInfoBean.setModuleType(Modules.valueOf(schedulerDto.getSourceModule()).getModuleType());
+        taskScheduleBeanInfoBean.setTaskType(taskType);
+        taskScheduleBeanInfoBean.scheduleScanStartDate(schedulerDto.getStartDateTime());
+        taskScheduleBeanInfoBean.setScheduleScanEndDate(SCHEDULE_END_DATE);
+        taskScheduleBeanInfoBean.setScheduleType(schedulerDto.getScanType());
+        taskScheduleBeanInfoBean.setTaskIds(taskIdList);
+        taskScheduleBeanInfoBean.setScheduleDay(schedulerDto.getScheduleDay());
+        taskScheduleBeanInfoBean.setTimeZone("UTC");
+
+        logger.info("Task info bean to controller: " + taskScheduleBeanInfoBean);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/SchedulerService.java L145
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1175064611>*